### PR TITLE
Ebnt 128 xgboost

### PR DESCRIPTION
**Pull request by mike0sv**

### CODE REVIEW

File: src/ebonite/ext/ext_loader.py [hunk 1]
```diff
@@ -60,7 +60,8 @@ class ExtensionLoader:
         Extension('ebonite.ext.flask', ['flask', 'flaskgger']),
         Extension('ebonite.ext.sqlalchemy', ['sqlalchemy']),
         Extension('ebonite.ext.s3', ['boto3']),
-        Extension('ebonite.ext.imageio', ['imageio'])
+        Extension('ebonite.ext.imageio', ['imageio']),
+        Extension('ebonite.ext.xgboost', ['xgboost'])
```
Comment (i-a-andrianov) [inline src/ebonite/ext/ext_loader.py]: I don't thnk that XGBoost should be mandatory extension

File: src/ebonite/ext/xgboost/dataset.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+class DMatrixDatasetType(DatasetType):
+    type = 'xgboost_dmatrix'
+
+    real_type = xgboost.DMatrix
+
+    def get_spec(self) -> ArgList:
+        pass
+
+    def serialize(self, instance: xgboost.DMatrix) -> list:
+        raise RuntimeError('DMatrixDatasetType does not support serialization')
```
Comment (i-a-andrianov) [inline src/ebonite/ext/xgboost/dataset.py]: Please describe reasons of such limitation

File: tests/ext/sklearn/test_model.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+def wrapper(model):
+    return ModelAnalyzer.analyze(model)
+
+
+def test_hook(wrapper):
+    assert isinstance(wrapper, SklearnModelWrapper)
+
+
+def test_wrapper__predict(wrapper, data):
+    data = [[1, 2, 3]]
```
Comment (i-a-andrianov) [inline tests/ext/sklearn/test_model.py]: duplicates fixture

File: tests/ext/sklearn/test_model.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+    prediction = wrapper.predict(data)
+
+    assert len(prediction) == len(data)
+
+
+def test_wrapper__dump_load(tmpdir, wrapper):
+    data = [[1, 2, 3]]
```
Comment (i-a-andrianov) [inline tests/ext/sklearn/test_model.py]: ignores fixture

File: tests/ext/xgboost/test_dataset.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+@pytest.fixture
+def dtype(dmatrix):
+    return DatasetAnalyzer.analyze(dmatrix)
+
+
+def test_hook(dtype):
+    assert issubclass(dtype, DMatrixDatasetType)
+
+
+def test_serialize(dtype: DMatrixDatasetType()):
```
Comment (i-a-andrianov) [inline tests/ext/xgboost/test_dataset.py]: what does `()` mean in this context?

File: tests/ext/xgboost/test_dataset.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    payload = [1, 2, 3, 4]
+    dmatrix = xgboost.DMatrix(payload)
+    with pytest.raises(RuntimeError):
+        dtype.serialize(dmatrix)
+
+
+def test_deserialize(dtype: DMatrixDatasetType()):
```
Comment (i-a-andrianov) [inline tests/ext/xgboost/test_dataset.py]: what does `()` mean in this context?

File: tests/ext/xgboost/test_dataset.py (hunk 5 above)
Comment (mike0sv) [reply tests/ext/xgboost/test_dataset.py]: DMatrixDatasetType is a metaclass, kind of. So `()` means that we actually creating *type*

File: src/ebonite/ext/ext_loader.py (hunk 1 above)
Comment (mike0sv) [reply src/ebonite/ext/ext_loader.py]: done

File: src/ebonite/ext/xgboost/dataset.py (hunk 2 above)
Comment (mike0sv) [reply src/ebonite/ext/xgboost/dataset.py]: left a comment in code

File: tests/ext/sklearn/test_model.py (hunk 3 above)
Comment (mike0sv) [reply tests/ext/sklearn/test_model.py]: fixed

File: tests/ext/sklearn/test_model.py (hunk 4 above)
Comment (mike0sv) [reply tests/ext/sklearn/test_model.py]: fixed

**Pull request merged**